### ISSUE TITLE
Half forms are not to be used in Bengali

### ISSUE THREAD

**Issue by sridatta1** [notofonts/bengali#10]
### Title 
Half forms are not to be used in Bengali
### Font 
Noto Sans Bengali & Noto Serif Bengali

### Where the font came from, and when
Tested on https://fonts.google.com/noto/fonts site:
June 26, 2022

### Issue
Half forms are not the feature of Bengali script, they are not found traditonally. For supporting arbitrary conjunct sequences, some of font makers have tried inventing Devanagari-like half forms for Bengali. The conjuncts with smaller half forms look very incorrect.
So for the other conjuncts where there are not ligatures in the font, either the ligatures can be added wherever it's necessary or just the virama should be used for the fallback.
e.g.
Text: স্স  গ্ঘ ক্শ প্ফ ক্খ  ম্হ 
  2. Observed results
  
![image](https://user-images.githubusercontent.com/23263723/175826552-5957c4d0-73ba-4b60-ac6c-e219acac0c12.png)

Tiro Bangla font
![image](https://user-images.githubusercontent.com/23263723/175826587-ee89c260-2fd2-4745-b705-a1405be7372d.png)

Also the following comment on Bengali half-forms can be looked into
https://github.com/notofonts/bengali/issues/1#issuecomment-1160653599
https://github.com/notofonts/bengali/issues/1#issuecomment-1160653600

**Comment by simoncozens** [comment]
>  For supporting arbitrary conjunct sequences, some of font makers have tried inventing Devanagari-like half forms for Bengali. The conjuncts with smaller half forms look very incorrect.

This may be, I'm afraid, one of those things which starts out "wrong" but becomes right because everyone is doing it and people get used to seeing it. Checking the fonts on Google Fonts, ITF and Tiro use viramas but everyone else uses half forms. (Important context here: I *really* respect ITF and Tiro, and I think they are very likely to be correct. I'm not talking about correctness, just that they are outliers.)

I suspect if we remove the half forms, we will start getting issues asking where they're gone.

Maybe we can move them to some discretionary feature;  perhaps a stylistic set. But I think I need more advice before making such a visible change.

**Comment by joanaranito** [comment]
> , I'm afraid, one of those things which starts out "wrong" but becomes right because everyone is doing it and people get used to seeing it.

> ITF and Tiro use viramas but everyone else uses half forms. (Important context here: I really respect ITF and Tiro, and I think they are very likely to be correct. I'm not talking about correctness, just that they are outliers.

At Universal Thirst we also don't include Bangla half-forms. Linear half-forms (like in Devanagari) don't exist in Bangla, but are becoming popular because they come in the predefined Bangla set in Glyphs; and because the pre-defined Bangla engineering in Glyphs requires them. But that doesn't make them right or a feature of the Bangla script.

Regarding the concern about removing the half-forms, I'd argue that everyone can read and recognise a full form + virama, whereas incorrectly implemented half-forms not only look wrong, they can even can cause confusion in reading. Some of the half-forms are just shrunken versions of the full form (like ka, for example - see image below). But because the following character in the unsupported conjunct doesn't stack below the ka half-form, the ka half-form can easily be mistaken by a full-form ka. 

<img width="378" alt="Screenshot 2023-11-23 at 13 18 35" src="https://github.com/notofonts/bengali/assets/4913245/f9488263-9417-404d-a2de-a3dbe57f95a2">

"Everyone is doing it" doesn't seem like a good enough reason to change how a script works in type vs how it works in writing :)

**Comment by simoncozens** [comment]
Fair enough, I agree with your arguments for removing them. I will work on that. But this:

> "Everyone is doing it" doesn't seem like a good enough reason to change how a script works in type vs how it works in writing :)

I'm not sure about in general. Typographic practices can become mainstream, and from there become expected, and from there inadvertently cause the evolution of writing systems. (For example, how many Tamil typefaces adopted the Latin stroke contrast from the Western mission presses.) This is one of the reasons why Noto is so important to me; if we get stuff wrong, particularly with minority scripts which don't have too many other fonts available for them, we could end up setting a new, corrupt norm...

**Comment by r12a** [comment]
Are we talking about all conjuncts that use 'half forms', or just the ones mentioned?  As far as i can see, it is only the conjuncts listed in the initial comment that are converted to sequences with a visible virama in the Tiro Bangla font.

Examples of side-by-side combinations of glyphs to make conjuncts in the Noto Serif Bengali font. The combinations mentioned in the initial comment are those on the 3rd line:
<img width="690" alt="Image" src="https://github.com/user-attachments/assets/504b1022-a120-4626-ad7f-4ad9ab8132fd" />

The same list of conjuncts in Tiro Bangla:
<img width="717" alt="Image" src="https://github.com/user-attachments/assets/0093340c-3b8e-4a23-a4ff-faf4a02f39dd" />

List as characters:
প্ট প্স ব্ব ব্দ দ্দ ধ্ব ড্ড চ্চ চ্ছ চ্ঞ জ্ব জ্জ জ্ঝ জ্জ্ব ষ্প ষ্ফ শ্চ শ্ছ স্প স্ফ স্ট ল্ফ ম্প ম্ফ ন্দ ন্স ণ্ট ণ্ঠ ণ্ঢ ঙ্খ ঙ্ম ল্ট ল্ড 
স্স গ্ঘ ক্শ প্ফ ক্খ ম্হ

**Comment by sridatta1** [comment]
Only  the ones which are not having traditional forms in the printed books/ manuscripts -
E.g. all those rendered with virama ib Tiro font.
Where fonts insert these invented half forms

**Comment by sridatta1** [comment]
স্স গ্ঘ প্ফ ক্খ ম্হ
On these are written the document in #25 can be checked
